ImageTaggingApp.Console.App/Services/ImageTaggingService.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using ImageTaggingApp.Console.App.APIs;
+using ImageTaggingApp.Console.App.Entities;
+
+namespace ImageTaggingApp.Console.App.Services {
+    public interface IImageTaggingService {
+        IList<Image> Tag(IReadOnlyCollection<string> imagesPaths);
+    }
+
+    public class ImageTaggingService : IImageTaggingService {
+        private readonly IImageTaggingApi _imageTaggingApi;
+
+        public ImageTaggingService(IImageTaggingApi imageTaggingApi) {
+            _imageTaggingApi = imageTaggingApi;
+        }
+
+        public IList<Image> Tag(IReadOnlyCollection<string> imagesPaths) {
+            if (imagesPaths.Count == 0) {
+                throw new ArgumentException("Image paths count is equal to zero.", nameof(imagesPaths));
+            }
+            var taggedImages = new List<Image>();
+            foreach (var imagePath in imagesPaths) {
```

Comment:
several notes here:
1) if there will be 1000 images it will take long time, so
    - it would be nice to see progress
    - it would be nice to have ability to cancel
    - I think it would be better to save results more often or cache them temporary to disk, because you'll use your paid/free API calls and then fail to save to database and it will be double fail :)
2) Wondering if we need to track API calls limit here and gracefully handle situation when limit is reached or check if there will be enough limit before starting to process (may be provide user to choose how to handle)